### PR TITLE
Fix indy lambdas with extension and context receivers

### PR DESCRIPTION
**Pull request by mcpiroman**
KT-53957 Fixed
KT-52874 likely fixed

Parameters were ordered incorrectly: [extension receiver, context receivers, regular parameters] instead of [context receivers, extension receiver, regular parameters].

Wouldn't it be reasonable to have a separate list of context parameters (instead of `contextReceiverParametersCount`), the same as `extensionReceiverParameter` is separate in IR?

### CODE REVIEW

**Comment by udalov** [comment]
> Wouldn't it be reasonable to have a separate list of context parameters (instead of `contextReceiverParametersCount`), the same as `extensionReceiverParameter` is separate in IR?

As a part of [KT-53696](https://youtrack.jetbrains.com/issue/KT-53696), we're rather considering merging all parameters into one list, so that many places like this won't need to handle three kinds of parameters. But it's still being discussed.

File: compiler/fir/fir2ir/tests-gen/org/jetbrains/kotlin/test/runners/codegen/FirBlackBoxCodegenTestGenerated.java [hunk 1]
```diff
@@ -6,10 +6,10 @@
 package org.jetbrains.kotlin.test.runners.codegen;
 
 import com.intellij.testFramework.TestDataPath;
-import org.jetbrains.kotlin.test.util.KtTestUtil;
```
Comment (strangepleasures) [inline compiler/fir/fir2ir/tests-gen/org/jetbrains/kotlin/test/runners/codegen/FirBlackBoxCodegenTestGenerated.java:9]: Please, don't modify the order of imports in the auto-generated files; that causes some build checks to fail. Make sure that you don't have "Before Commit" actions enabled.

**Review by strangepleasures — CHANGES_REQUESTED** [review]
Thanks for contributing your fix. Please revert the changes in the import sections of the auto-generated files. Otherwise looks great.

**Comment by strangepleasures** [comment]
@mcpiroman Please revert formatting in FirBlackBoxCodegenTestGenerated and IrBlackBoxCodegenTestGenerated as well. Sorry for not being explicit about that.

**Comment by Ponchest** [comment]
Dejar de seguir mi información ya r colocado denuncio

**Pull request merged**